command/agent/command.go

```diff
@@ -1081,8 +1081,12 @@ func (c *Command) handleReload(config *Config) (*Config, error) {
 		errs = multierror.Append(errs, fmt.Errorf("Failed unloading checks: %s", err))
 		return nil, errs
 	}
+	if err := c.agent.unloadMetadata(); err != nil {
```

Comment:
I'm not sure that'll work though - there are other error paths in here as well, so we need a way to save the old meta, unload + reload, and restore the old meta if anything goes wrong with any part of the reload.